File: greedy/1221_split_a_string_in_balanced_strings/1221_split_a_string_in_balanced_strings.py

class Solution:
    def balancedStringSplit(self, s: str) -> int:
        split = 0
        unbalance = 0
        for i in s:
            unbalance += 1 if i == 'R' else -1    # 'R' = +1, 'L' = -1
            if not unbalance:   # if unbalance == 0
                split += 1
        return split


s = "RLRRLLRLRL"
res = Solution().balancedStringSplit(s)
print(res)

File: greedy/1221_split_a_string_in_balanced_strings/1221_split_a_string_in_balanced_strings.go
package main

import "fmt"

func balancedStringSplit(s string) int {
	var stack int
	var out int
	for i := range s {
		if s[i] == 'R' {
			stack++
		} else {
			stack--
		}
		if stack == 0 {
			out++
		}
	}
	return out
}

func main() {
	s := "RLRRLLRLRL"
	res := balancedStringSplit(s)
	fmt.Println(res)
}
